### PR TITLE
test/librados: don't release handler if set_pg_num failed

### PR DESCRIPTION
**Pull request by hjwsm1989**
The handler will be released by RadosTestNS::TearDownTestCase()
before process exit.

Signed-off-by: huangjun <huangjun@xsky.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by hjwsm1989** [comment]
@badone  please take a review

File: src/test/librados/test_common.cc [hunk 1]
```diff
@@ -132,7 +132,6 @@ std::string with_healthy_cluster(rados_t* cluster,
       throw pool_op_error{pool_name, "wait_for_healthy", r};
     }
   } catch (const pool_op_error& e) {
-    rados_shutdown(*cluster);
```
Comment (tchaikov) [inline src/test/librados/test_common.cc]: also, why do you think this fixes https://tracker.ceph.com/issues/42706 ? how is this call related to the return value of `with_healthy_cluster()`?

Comment (hjwsm1989) [reply src/test/librados/test_common.cc]: set_pg_num will call with_healthy_cluster, and if set_pg_num failed, it will raise pool_op_error,
that will call rados_shutdown, and ASSERT_TRUE(set_pg_num(&s_cluster, pool_name, 11).empty());
will fail, then will call RadosTestNS::TearDown(), but in this test suite it will clean the objects first, then call RadosTestNS::cleanup_all_objects().

**Comment by hjwsm1989** [comment]
@tchaikov any tips?

**Comment by hjwsm1989** [comment]
@liewegas @badone @tchaikov  please help review

File: src/test/librados/test_common.cc (hunk 1 above)
Comment (tchaikov) [reply src/test/librados/test_common.cc]: if a `pool_op_error` is thrown, `set_pg_num()` will return a non-empty string. so the test still fails with your change. am i missing something?

**Comment by badone** [comment]
@hjwsm1989 Thanks for submitting this. I'll try and take a look in the next day or so.

File: src/test/librados/test_common.cc (hunk 1 above)
Comment (hjwsm1989) [reply src/test/librados/test_common.cc]: it's ok to fail, the issue is it shouldn't segfault.

Comment (tchaikov) [reply src/test/librados/test_common.cc]: i appreciate this fix. but could you please remove "Fixes: https://tracker.ceph.com/issues/42706" from the commit message. i still don't think this change addresses the failure describe by it.

**Comment by hjwsm1989** [comment]
@tchaikov updated

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2020-01-07_01:48:07-rados-wip-kefu-testing-2020-01-07-0032-distro-basic-smithi/

failures fixed / tracked by

- https://github.com/ceph/ceph/pull/32517
- https://github.com/ceph/ceph/pull/32518
- https://github.com/ceph/ceph/pull/32519
- https://tracker.ceph.com/issues/43431
- https://tracker.ceph.com/issues/43492
- https://tracker.ceph.com/issues/43494
